tfjs-backend-webgpu/src/webgpu_util.ts

```diff
@@ -58,6 +58,30 @@ export function computeDispatch(
   ];
 }
 
+export function computeWorkGroupSizeForConv2d(
+    layout: {x: number[], y?: number[], z?: number[]},
+    outputShape: number[]): [number, number, number] {
+  const dim1 = arrayProduct(layout.y.map(d => outputShape[d]));
+  // TODO(jiajia.qin@intel.com): More fine tune based on outputShape.
+  if (dim1 <= 4) {
```

Comment:
Don't we want to additionally check that dim0 is significantly bigger than dim1?

Reply:
I think it's not needed. The conv2d(1, 128, 128, 4) benchmark already proves that it runs very fast when dim0 is significantly bigger than dim1. I think the root cause of the improvement is that dim1 doesn't need 16 threads. Otherwise you will waste at least 12 threads in each work group for nothing (out of bounds). If we set y dimension to 4, the hardware can transfer the other 12 threads to x dimension to do the real work.

Reply:
Even if dim0 is not significantly bigger than dim1, it doesn't matter. We already fully used y dimension. We can add more rules later. Like dim0 <=4, maybe need to do the similar change. And other situations, like if 4< dim0 < 8, 8 may be the best choice for x dimension.